elfpy/time/time.py

```diff
@@ -154,34 +154,97 @@ def time_to_days_remaining(time_remaining: float, time_stretch: float = 1, norma
     return normed_days_remaining * normalizing_constant
 
 
+TimeUnit = Literal["seconds", "minutes", "hours", "days", "years"]
+
+
 @dataclass
 class BlockTimeFP:
-    r"""Global time."""
+    r"""State class for tracking block timestamps and global time"""
 
-    time: FixedPoint = FixedPoint(0)  # time in years
-    block_number: FixedPoint = FixedPoint(0)
-    step_size: FixedPoint = FixedPoint("1.0") / FixedPoint("365.0")  # defaults to 1 day
+    _time: FixedPoint = FixedPoint(0)
+    _block_number: FixedPoint = FixedPoint(0)
+    _step_size: FixedPoint = FixedPoint("1.0") / FixedPoint("365.0")  # defaults to 1 day
+    time_unit: str = "years"
 
-    @property
-    def time_in_seconds(self) -> FixedPoint:
-        """1 year = 31,556,952 seconds"""
-        return self.time * FixedPoint("31_556_952.0")
+    def __post_init__(self):
+        if self.time_unit != "years":
+            raise ValueError(f"Only `years` is supported for {self.time_unit}")
 
     def tick(self, delta_years: FixedPoint) -> None:
         """ticks the time by delta_time amount"""
-        self.time += delta_years
+        self._time += delta_years
 
     def step(self) -> None:
         """ticks the time by step_size"""
-        self.time += self.step_size
+        self._time += self.step_size
+
+    def time_conversion(self, unit: TimeUnit = "seconds") -> FixedPoint:
+        """Convert time to different units
+
+        .. todo:: we will need to add conditions for self.time_unit in each conversion type
+        """
+        if unit == "seconds":  # 31,556,952 seconds in a year
+            return self.time * FixedPoint("31_556_952.0")
+        if unit == "minutes":  # 525,600 moments so dear
+            return self.time * FixedPoint("525_600.0")
+        if unit == "hours":  # 8,760 hours in a year
+            return self.time * FixedPoint("8_760.0")
+        if unit == "days":  # 365 days in a year
+            return self.time * FixedPoint("365.0")
+        if unit == "years":  # 1 years in a year
+            return self.time * FixedPoint("1.0")
+        raise NotImplementedError(f"time {unit=} is not yet supported.")
+
+    @property
+    def time(self):
+        """Get the time"""
+        return self._time
+
+    @time.setter
+    def time(self, value):
+        """Private time attribute and time property cannot be set directly,
+        since we want to check types using a setter."""
+        raise AttributeError("time is a read-only attribute; use `set_time()` to adjust")
 
     def set_time(self, time: FixedPoint) -> None:
```

Comment:
could have another function that takes in a `TimeUnit`, converts to years then sets the time in years. that is:
```py
def set_time(self, time: FixedPoint, time_unit: TimeUnit) -> None:
	self.set_time(_convert_time(from=time_unit, to="years"))
```

Reply:
the conversion function only goes from years to `time_unit`. It's not any unit -to- any unit. And as above, I didn't want to write all of that out until we decide how we want the time refactor to look.